### PR TITLE
Tune sentencepiece alphas

### LINKED ISSUE #411: Investigate issues with sentencepiece sampling
Apparently setting `--sentencepiece-alphas 0.1` for marian ruins training for en-ru. It wasn't the case for en-lt.

It was added to improve robustness based on the recommendations from [this paper ](https://arxiv.org/pdf/2311.14838.pdf). 

Sentencepiece [readme section](https://github.com/google/sentencepiece#subword-regularization-and-bpe-dropout). 
Subword regularization [paper](https://arxiv.org/pdf/1804.10959.pdf).

I noticed that the same value was used in HPLT [configs](https://github.com/hplt-project/bitextor-mt-models/blob/a2f3229816d6f1f74f56f6445acd60a63f913797/glg-eng/teacher/marian.config.gl-en.yml#L14).

We should investigate how exactly setting alphas affects robustness, what values should be set and whether they are language dependent.

There might be another bug somewhere but the main difference between the normal training and diverged one was removing this option:
Screenshot 2024-01-30 at 2 02 07 PM

Screenshot 2024-01-30 at 2 02 01 PM

### PR DESCRIPTION
**Pull request by eu9ene**
fixes #411 

+ move to configs not to affect the training scripts when adjusting and tune separately for different models if needed

[skip ci]

### CODE REVIEW

File: docs/training-guide.md [hunk 1]
```diff
@@ -158,9 +160,26 @@ marian-args:
...
-Make sure to set `optimizer-delay` so that GPU devices * optimizer-delay = 8.
+#### Optimizer delay
+Make sure to set `optimizer-delay` so that `GPU devices * optimizer-delay = 8`.
 It makes training more stable.
 
+#### Subword regularization
+```
+sentencepiece-alphas: 0.5
+```
+Sentenpiece alphas control the alpha parameter in subword sampling for the unigram model.
```
Comment (gregtatum) [inline docs/training-guide.md:171]:
```suggestion
SentencePiece alphas control the alpha parameter in subword sampling for the unigram model. 
```

File: docs/training-guide.md [hunk 2]
```diff
@@ -158,9 +160,26 @@ marian-args:
...
 It makes training more stable.
 
+#### Subword regularization
+```
+sentencepiece-alphas: 0.5
+```
+Sentenpiece alphas control the alpha parameter in subword sampling for the unigram model. 
+It improves robustness of the model, especially for unseen domains. 
+
+If not specified, Marian does not run Sentencepiece sampling (corresponds to `alpha=1`).
```
Comment (gregtatum) [inline docs/training-guide.md:174]:
```suggestion
If not specified, Marian does not run SentencePiece sampling (corresponds to `alpha=1`). 
```

File: docs/training-guide.md [hunk 3]
```diff
@@ -158,9 +160,26 @@ marian-args:
...
+Sentenpiece alphas control the alpha parameter in subword sampling for the unigram model. 
+It improves robustness of the model, especially for unseen domains. 
+
+If not specified, Marian does not run Sentencepiece sampling (corresponds to `alpha=1`). 
+Lower values (`0.1`, `0.2`) increase randomization and might benefit lower resource languages with less diverse datasets. 
+However, the model might not train at all if the alpha is too low. 
+The recommended value to start with is `0.5`.
+
+More details:
+- [Sentecepiece readme](https://github.com/google/sentencepiece?tab=readme-ov-file#subword-regularization-and-bpe-dropout)
```
Comment (gregtatum) [inline docs/training-guide.md:180]:
```suggestion
- [SentencePiece readme](https://github.com/google/sentencepiece?tab=readme-ov-file#subword-regularization-and-bpe-dropout)
```

File: docs/training-guide.md [hunk 4]
```diff
@@ -158,9 +160,26 @@ marian-args:
     early-stopping: 20
 ```
 
-Make sure to set `optimizer-delay` so that GPU devices * optimizer-delay = 8.
+#### Optimizer delay
+Make sure to set `optimizer-delay` so that `GPU devices * optimizer-delay = 8`.
 It makes training more stable.
 
+#### Subword regularization
```
Comment (gregtatum) [inline docs/training-guide.md:167]: Praise: Thanks for the docs!

**Review by gregtatum — APPROVED** [review]
